hoodie-utilities/src/main/java/com/uber/hoodie/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -235,6 +237,30 @@ public KafkaOffsetGen(TypedProperties props) {
     return offsetRanges;
   }
 
+  /***
+   * check up checkpoint offsets is valid or not, if true,  return checkpoint offsets, else return earliest offsets
+   * @param cluster,
+   * @param checkpointOffsets
+   * @param topicPartitions
+   * @return fromOffset
+   */
+  private HashMap<TopicAndPartition, KafkaCluster.LeaderOffset> checkupValidOffsets(KafkaCluster cluster,
+                                                                                    HashMap<TopicAndPartition, KafkaCluster.LeaderOffset> checkpointOffsets,
+                                                                                    Set<TopicAndPartition> topicPartitions ) {
+    java.util.Set<TopicAndPartition> partitions = checkpointOffsets.keySet();
+    HashMap<TopicAndPartition, KafkaCluster.LeaderOffset> fromOffsets;
+    fromOffsets = new HashMap(ScalaHelpers.toJavaMap(
+            cluster.getEarliestLeaderOffsets(topicPartitions).right().get()));
+    for (TopicAndPartition partition: partitions) {
+      if (checkpointOffsets.get(partition).offset() < fromOffsets.get(partition).offset() ) {
```

Comment:
how does this handle new partitions being added since the last checkpoint? in that case, you won't find the partition in the `checkpointOffsets` and it will keep crashing with NPE?